### PR TITLE
Fueled pack

### PR DESCRIPTION
**Pull request by bkerns724**
Some people just want to watch the Spire burn.

### CODE REVIEW

**Review by modargo — CHANGES_REQUESTED** [review]
Looks pretty good. We already discussed design and balance for most of the cards and the redesigned cards look reasonable, though I'm still a bit concerned about Burn Bright. 

Other than that, the only significant issue I saw was with `PlayWithFireAction`. There's also a few smaller issues I left comments about.

Once those issues are resolved this should be ready to merge!

File: src/main/java/thePackmaster/actions/fueledpack/PlayingWithFireAction.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        this.num = num;
+    }
+
+    public void update() {
+        playSFX();
+        Consumer<Integer> dieConsumer = dieFace -> {
+            if (dieFace > IGNITE_THRESHOLD)
+                applyToSelfTop(new IgnitePower(adp(), 1));
+        };
+        att(new RollAction(sides, 0, dieConsumer, false));
```
[the remark this answers is not in the document]
Comment (bkerns724) [reply src/main/java/thePackmaster/actions/fueledpack/PlayingWithFireAction.java:31]: I got a pull request for that accepted for the features/expansionpacks branch, but it isn't on the main branch.  Should I submit a pull request for that branch or is expansion packs branch getting merged in?

File: src/main/resources/anniv5Resources/localization/eng/fueledpack/Cardstrings.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "DESCRIPTION": "${ModID}:Consume a card to deal !D! damage and apply !M! ${ModID}:Ignite to ALL enemies."
+  },
+  "${ModID}:PlayingWithFire": {
+    "NAME": "Playing With Fire",
+    "DESCRIPTION": "Roll !M! ${ModID}:D8. NL Gain 1 ${ModID}:Ignite for each 7 or 8 you roll."
+  },
+  "${ModID}:PhoenixHeart": {
+    "NAME": "Phoenix Heart",
+    "DESCRIPTION": "Exhaust all *Hot *Ash in your hand. NL Add a random card into your hand for each Exhausted card.",
+    "UPGRADE_DESCRIPTION": "Exhaust all *Hot *Ash in your hand. NL Add a random Upgraded card into your hand for each Exhausted card."
```
[the remark this answers is not in the document]
Comment (bkerns724) [reply src/main/resources/anniv5Resources/localization/eng/fueledpack/Cardstrings.json:35]: I failed to update the card properly, the description is what its supposed to do.

File: src/main/java/thePackmaster/cards/fueledpack/BurningStrike.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.DrawCardAction;
+import com.megacrit.cardcrawl.actions.common.ExhaustAction;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+import static thePackmaster.util.Wiz.atb;
+
+public class BurningStrike extends AbstractFueledCard {
```
[the remark this answers is not in the document]
Comment (bkerns724) [reply src/main/java/thePackmaster/cards/fueledpack/BurningStrike.java:12]: Several playtests.  The old version just didn't feel like it had enough oomph, I often regretted taking it in act 1 where you want to be picking up your damage commons.

File: src/main/java/thePackmaster/actions/fueledpack/PlayingWithFireAction.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (erasels) [reply src/main/java/thePackmaster/actions/fueledpack/PlayingWithFireAction.java:31]: That can't be. The feature branch is merged into main. You should recompile the packmaster jar locally so you're working with the newest version

File: src/main/java/thePackmaster/cards/fueledpack/FireOfLife.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package thePackmaster.cards.fueledpack;
+
+import com.evacipated.cardcrawl.mod.stslib.actions.tempHp.AddTemporaryHPAction;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+import static thePackmaster.util.Wiz.atb;
+
+public class FireOfLife extends AbstractFueledCard {
```
[the remark this answers is not in the document]
Comment (bkerns724) [reply src/main/java/thePackmaster/cards/fueledpack/FireOfLife.java:10]: I'm glad you like it

**Comment by bkerns724** [comment]
I believe this is now ready once I know what to do with the dice PR for Packmaster repo.

File: src/main/java/thePackmaster/actions/fueledpack/PlayingWithFireAction.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (modargo) [reply src/main/java/thePackmaster/actions/fueledpack/PlayingWithFireAction.java:31]: I see the PR mentioned (https://github.com/erasels/PackmasterCharacter/pull/648/files) and it was merged. It looks to me like the issue is that it was merged _after_ the PR for the expansion packs branch was merged. So the changes are sort of marooned on a branch that's no longer being actively developed against.

Luckily this is easy to fix now that I see what happened. I've fixed this by merging the expansion packs feature branch again, see https://github.com/erasels/PackmasterCharacter/pull/649. Now if you pull in the most recent code from the Packmaster repo on the `main` branch and build that, it should work properly with this PR.

I've tested this myself locally and things seem to build and run properly now.

File: src/main/java/thePackmaster/cards/fueledpack/BurningStrike.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    private static final int DAMAGE = 9;
+    private static final int UPGRADE_DAMAGE = 1;
+    private static final int MAGIC = 1;
+    private static final int UPGRADE_MAGIC = 1;
+
+    public BurningStrike() {
+        super(ID, COST, TYPE, RARITY, TARGET);
+        baseDamage = DAMAGE;
+        baseMagicNumber = magicNumber = MAGIC;
+    }
```
Comment (modargo) [inline src/main/java/thePackmaster/cards/fueledpack/BurningStrike.java:28]: With the name, this should have the strike tag: `tags.add(CardTags.STRIKE);`

File: src/main/resources/anniv5Resources/images/cards/Conflaguration.png
[the remark this answers is not in the document]
Comment (modargo) [reply src/main/resources/anniv5Resources/images/cards/Conflaguration.png:1]: There's no Burninate image and no Infernal Blaze image in the file list for the PR, so I'm pretty sure those are still missing and need to be added.

I ran things locally to confirm this. You can see that those two cards are getting card roller art.
image

Maybe you have the correct images locally but didn't add them to Git?

[the remark this answers is not in the document]
Comment (bkerns724) [reply src/main/resources/anniv5Resources/images/cards/Conflaguration.png:1]: Fixed.  Unless some sort of cache or something is tricking me again.

**Comment by bkerns724** [comment]
I think this is ready.  Sorry if I missed something.

**Comment by modargo** [comment]
> I think this is ready. Sorry if I missed something.

Looks like Burning Strike is still missing the strike tag, but I'll just merge this and ad that in myself. I'm looking forward to playing with the release version of this pack!

**Pull request merged**